naresh/src/main/java/com/lftechnology/phpjava/ems/dao/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,236 @@
+            value = employee.getAddress();
+            sql.append(" address = ?");
+        } else if (employee.getDepartment() != null && !employee.getDepartment().isEmpty()) {
+            update = true;
+            value = employee.getDepartment();
+            sql.append(" department = ?");
+        }
+        if (update) {
+            sql.append(" WHERE user_id = ? ");
+            stmt = conn.prepareStatement(sql.toString());
+            stmt.setString(1, value);
+            stmt.setInt(2, employee.getUserId());
+            return stmt.executeUpdate();
+        }
+        return 0;
+    }
+
+    /**
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     * @param employee
+     * @return
+     * @throws SQLException
+     */
+    @Override
+    public int delete(Employee employee) throws SQLException {
+        return 0;
+    }
+
+
+    /**
+     * Find the employee by fullname
+     *
+     * @param fullName
+     * @return Employee
+     * @throws SQLException
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    public List<Employee> findByFullName(String fullName) throws SQLException {
+
+        String sql = "SELECT * FROM employee WHERE fullname = ?";
+
+        stmt = conn.prepareStatement(sql);
+        stmt.setString(1, fullName);
+        return findBy(stmt);
+    }
+
+    /**
+     * Find the employee by address
+     *
+     * @param address
+     * @return Employee
+     * @throws SQLException
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    public List<Employee> findByAddress(String address) throws SQLException {
+
+        String sql = "SELECT * FROM employee WHERE address = ?";
+
+        stmt = conn.prepareStatement(sql);
+        stmt.setString(1, address);
+        return findBy(stmt);
+    }
+
+    /**
+     * Find the employee by department
+     *
+     * @param department
+     * @return Employee
+     * @throws SQLException
+     */
+
+    public List<Employee> findByDepartment(String department) throws SQLException {
+
+        String sql = "SELECT * FROM employee WHERE department = ?";
+
+        stmt = conn.prepareStatement(sql);
+        stmt.setString(1, department);
+        return findBy(stmt);
+    }
+
+    /**
+     * Find the employee by role
+     *
+     * @param
+     * @return Employee
+     * @throws SQLException
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+
+    public List<Employee> findByRole(Role role) throws SQLException {
+
+        String sql = "SELECT * FROM employee WHERE role = ?";
+
+        stmt = conn.prepareStatement(sql);
+        stmt.setString(1, role.toString());
+        return findBy(stmt);
+    }
+
+    /**
+     * Find the employee by role
+     *
+     * @param
+     * @return Employee
+     * @throws SQLException
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+
+    public List<Employee> findByUserId(int userId) throws SQLException {
```

Comment:
If you are finding by userId it should always return single Object. Change the method definition.